File: Algorithms/Data Structure dependent/heaps/top k/kClosest.java
class Solution {
    class Point implements Comparable<Point> {
        public int x,y;
        public Point(int x, int y){
            this.x = x;
            this.y = y;
        }
        public int compareTo(Point p){
            return (int)(p.getDist() - this.getDist());
        }
        public double getDist(){
            return Math.pow(x, 2) + Math.pow(y, 2);
        }
    }

    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<Point> pq = new PriorityQueue<Point>();
        for(int[] point: points){
            Point p = new Point(point[0], point[1]);
            pq.add(p);
            if(pq.size()> k) pq.poll();
        }

        int[][] out = new int[k][2];
        while(pq.size() != 0){
            k--;
            Point p = pq.poll();
            out[k][0] = p.x;
            out[k][1] = p.y;
        }

        return out;

    }
}

File: Algorithms/Data Structure dependent/heaps/top k/kClosest.py
from math import sqrt
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:

        pq = []

        points = [(-sqrt(x**2+y**2),x,y) for x,y in points]
        for val in points:
            heappush(pq, val)
            if len(pq) > k: heappop(pq)

        return [[x, y] for _, x, y in pq]